### ISSUE TITLE
ImportError: cannot import name 'preserve_channel_dim' from 'albucore.utils'

### ISSUE THREAD

**Issue by margo2130** [Gourieff/comfyui-reactor-node#432]
### First, confirm

- [X] I have read the [instruction](https://github.com/Gourieff/comfyui-reactor-node/blob/main/README.md) carefully
- [X] I have searched the existing issues
- [X] I have updated the extension to the latest version

### What happened?

hello,
i need to install ReActor Node for ComfyUI but it always get me this error: ((IMPORT FAILED) ReActor Node for ComfyUI)
![369617813-ed12c230-ea47-4193-b8f9-6ee40684cf12](https://github.com/user-attachments/assets/6c581eb0-4b0f-41a0-982a-d6033d5e161d)

i do whatere i can.
even i did this:
https://github.com/Gourieff/comfyui-reactor-node#insightfacebuild
but i couldn't solve it. every time i face this error. i select "fix" button but not solve. please help me.

### Steps to reproduce the problem

i install this node and keep getting the error. i don't know what is the problem and how to solve it.

### Sysinfo

windows 10. rtx 4070 super 12gb.
the nodes:
[link](https://private-user-images.githubusercontent.com/80190186/368583815-e6ebf932-50da-45a6-8f31-f74f8182bc15.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NzkwMjgsIm5iZiI6MTcyNjk3ODcyOCwicGF0aCI6Ii84MDE5MDE4Ni8zNjg1ODM4MTUtZTZlYmY5MzItNTBkYS00NWE2LThmMzEtZjc0ZjgxODJiYzE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIyVDA0MTg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmMDZhNWY5NWU3ODc5ZGJmMzMzYTllODIwOGE0MzBmZDFmMzBlMWVjZTg1ODYwMTY5ZDZhNTYzMjkwODAzNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VKRnoi8_ZrW683ortDZDA5E3eMF-IO4p1QHtJI21FjI)

### Relevant console log

```Shell
F:\AIs\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build
[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies done.
** ComfyUI startup time: 2024-09-22 07:39:31.659602
** Platform: Windows
** Python version: 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
** Python executable: F:\AIs\ComfyUI_windows_portable\python_embeded\python.exe
** ComfyUI Path: F:\AIs\ComfyUI_windows_portable\ComfyUI
** Log path: F:\AIs\ComfyUI_windows_portable\comfyui.log

Prestartup times for custom nodes:
  10.3 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager-main

Total VRAM 12282 MB, total RAM 32509 MB
pytorch version: 2.4.1+cu121
WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.0.1+cu118 with CUDA 1108 (you have 2.4.1+cu121)
    Python  3.11.5 (you have 3.11.9)
  Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
  Memory-efficient attention, SwiGLU, sparse and more won't be available.
  Set XFORMERS_MORE_DETAILS=1 for more details
F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\xformers\ops\swiglu_op.py:106: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_fwd
F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\xformers\ops\swiglu_op.py:127: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_bwd
xformers version: 0.0.22
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4070 SUPER : cudaMallocAsync
Using pytorch cross attention
[Prompt Server] web root: F:\AIs\ComfyUI_windows_portable\ComfyUI\web
Adding extra search path checkpoints F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path configs F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path vae F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/VAE
Adding extra search path loras F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/Lora
Adding extra search path loras F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/LyCORIS
Adding extra search path upscale_models F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/ESRGAN
Adding extra search path upscale_models F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/RealESRGAN
Adding extra search path upscale_models F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/SwinIR
Adding extra search path embeddings F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\embeddings
Adding extra search path hypernetworks F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/hypernetworks
Adding extra search path controlnet F:\AIs\Stable Diffusion AUTOMATIC1111\stable-diffusion-webui\models/ControlNet
F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kornia\feature\lightglue.py:44: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @torch.cuda.amp.custom_fwd(cast_inputs=torch.float32)
### Loading: ComfyUI-Manager (V2.50.3)
### ComfyUI Revision: 2715 [364b69e9] | Released on '2024-09-21'
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
Traceback (most recent call last):
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1994, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-reactor-node\__init__.py", line 23, in <module>
    from .nodes import NODE_CLASS_MAPPINGS, NODE_DISPLAY_NAME_MAPPINGS
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-reactor-node\nodes.py", line 15, in <module>
    from insightface.app.common import Face
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\__init__.py", line 18, in <module>
    from . import app
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\app\__init__.py", line 2, in <module>
    from .mask_renderer import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\app\mask_renderer.py", line 4, in <module>
    import albumentations as A
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\__init__.py", line 6, in <module>
    from .augmentations import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\__init__.py", line 1, in <module>
    from .blur.functional import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\blur\__init__.py", line 1, in <module>
    from .functional import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\blur\functional.py", line 9, in <module>
    from albucore.utils import clipped, maybe_process_in_chunks, preserve_channel_dim
ImportError: cannot import name 'preserve_channel_dim' from 'albucore.utils' (F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albucore\utils.py)

Cannot import F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-reactor-node module for custom nodes: cannot import name 'preserve_channel_dim' from 'albucore.utils' (F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albucore\utils.py)
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\diffusers\utils\outputs.py:63: FutureWarning: `torch.utils._pytree._register_pytree_node` is deprecated. Please use `torch.utils._pytree.register_pytree_node` instead.
  torch.utils._pytree._register_pytree_node(
Traceback (most recent call last):
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1994, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-V-Express\__init__.py", line 1, in <module>
    from .nodes import NODE_CLASS_MAPPINGS, NODE_DISPLAY_NAME_MAPPINGS
  File "F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-V-Express\nodes.py", line 14, in <module>
    from insightface.app import FaceAnalysis
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\__init__.py", line 18, in <module>
    from . import app
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\app\__init__.py", line 2, in <module>
    from .mask_renderer import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\insightface\app\mask_renderer.py", line 4, in <module>
    import albumentations as A
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\__init__.py", line 6, in <module>
    from .augmentations import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\__init__.py", line 1, in <module>
    from .blur.functional import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\blur\__init__.py", line 1, in <module>
    from .functional import *
  File "F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albumentations\augmentations\blur\functional.py", line 9, in <module>
    from albucore.utils import clipped, maybe_process_in_chunks, preserve_channel_dim
ImportError: cannot import name 'preserve_channel_dim' from 'albucore.utils' (F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albucore\utils.py)

Cannot import F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-V-Express module for custom nodes: cannot import name 'preserve_channel_dim' from 'albucore.utils' (F:\AIs\ComfyUI_windows_portable\python_embeded\Lib\site-packages\albucore\utils.py)
Base directory: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip
Checkpoints path: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip\Wav2Lip\checkpoints
Model path: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip\Wav2Lip\checkpoints\wav2lip_gan.pth
Wav2Lip path added to sys.path: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip\wav2lip
Directory created or exists: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip\facedetection
Current directory: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip
Wav2Lip path: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip\wav2lip
Using cuda for inference.

Import times for custom nodes:
   0.0 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\websocket_image_save.py
   0.1 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_UltimateSDUpscale
   0.3 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_bitsandbytes_NF4
   0.6 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager-main
   0.7 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-VideoHelperSuite
   1.5 seconds (IMPORT FAILED): F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfyui-reactor-node
   1.7 seconds (IMPORT FAILED): F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-V-Express
   3.6 seconds: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_wav2lip

Starting server

To see the GUI go to: http://127.0.0.1:8188
FETCH DATA from: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager-main\extension-node-map.json [DONE]
Update ComfyUI
CUSTOM NODE PULL: Success
Install: pip packages
Line is comment...skipping
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json [DONE]
FETCH DATA from: F:\AIs\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager-main\extension-node-map.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json [DONE]
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json [DONE]
```

### Additional information

my comfy ui version is : v0.2.2 Release.
i update all.

**Comment by IvayloPask** [comment]
<img width="1484" alt="Screenshot 2024-09-22 122528" src="https://github.com/user-attachments/assets/5bd68090-8ea0-4730-ad17-4f8cf739354e">
Same problem, and not only with this node !

**Comment by danger-electrodes** [comment]
Had, the same problem, I had to downgrade albucore to v 0.0.16 : 

you have to open a command prompt in python_embeded (YOUR_PATH_TO_COMFYUI/ComfyUI_windows_portable/python_embeded) and run the following command : 

python.exe -m pip install albucore==0.0.16

**Comment by IvayloPask** [comment]
> Had, the same problem, I had to downgrade albucore to v 0.0.16 :
> 
> you have to open a command prompt in python_embeded (YOUR_PATH_TO_COMFYUI/ComfyUI_windows_portable/python_embeded) and run the following command :
> 
> python.exe -m pip install albucore==0.0.16

This didn't work for me, any other ideas please !

**Comment by margo2130** [comment]
> Had, the same problem, I had to downgrade albucore to v 0.0.16 :
> 
> you have to open a command prompt in python_embeded (YOUR_PATH_TO_COMFYUI/ComfyUI_windows_portable/python_embeded) and run the following command :
> 
> python.exe -m pip install albucore==0.0.16

This worked for me. thank you.

**Comment by margo2130** [comment]
> > Had, the same problem, I had to downgrade albucore to v 0.0.16 :
> > you have to open a command prompt in python_embeded (YOUR_PATH_TO_COMFYUI/ComfyUI_windows_portable/python_embeded) and run the following command :
> > python.exe -m pip install albucore==0.0.16
> 
> This didn't work for me, any other ideas please !

i suggest you try this and update your comfy ui.
https://github.com/Gourieff/comfyui-reactor-node#insightfacebuild

**Comment by IvayloPask** [comment]
FAILED: ReActor Node for ComfyUI

**Comment by danger-electrodes** [comment]
> > Had, the same problem, I had to downgrade albucore to v 0.0.16 :
> > you have to open a command prompt in python_embeded (YOUR_PATH_TO_COMFYUI/ComfyUI_windows_portable/python_embeded) and run the following command :
> > python.exe -m pip install albucore==0.0.16
> 
> This didn't work for me, any other ideas please !

Can you show me the error log?

**Comment by IvayloPask** [comment]
now i saw this:
Cannot import C:\Users\ivayl\Desktop\ComfyUI\ComfyUI\custom_nodes\comfyui-reactor-node module for custom nodes: No module named 'insightface'

**Comment by danger-electrodes** [comment]
> now i saw this: Cannot import C:\Users\ivayl\Desktop\ComfyUI\ComfyUI\custom_nodes\comfyui-reactor-node module for custom nodes: No module named 'insightface'

You have to go back to the main github page, there is a section in troubleshooting with the title 

"I. (For Windows users) If you still cannot build Insightface for some reasons or just don't want to install Visual Studio or VS C++ Build Tools - do the following:"

You have to do what the other said, look at the version of python your reactor node local repo uses, I specify this because it may be different from the one you'd use, then download either insightface 3.10, 3.11, or 3.12 depending on the version that was printed in the console, then you will execute the wheel, make sure to restart comfyUI if it's running, it should solve the issue.

**Comment by Gourieff** [comment]
Fixed https://github.com/Gourieff/comfyui-reactor-node/commit/e5f5b98ea7eb8143f5e076f868374fbbf6e5bc57
Just install latest Albumentations https://github.com/albumentations-team/albumentations/releases/tag/1.4.16

**Comment by ForxBase** [comment]
> Fixed [e5f5b98](https://github.com/Gourieff/comfyui-reactor-node/commit/e5f5b98ea7eb8143f5e076f868374fbbf6e5bc57) Just install latest Albumentations https://github.com/albumentations-team/albumentations/releases/tag/1.4.16

Does this fix the problem of the error "When loading the graph, the following node types were not found:
ReActorFaceSwap
Nodes that have failed to load will show as red on the graph."? 

How to do it?

**Comment by Gourieff** [comment]
> > Fixed [e5f5b98](https://github.com/Gourieff/comfyui-reactor-node/commit/e5f5b98ea7eb8143f5e076f868374fbbf6e5bc57) Just install latest Albumentations https://github.com/albumentations-team/albumentations/releases/tag/1.4.16
> 
> Does this fix the problem of the error "When loading the graph, the following node types were not found: ReActorFaceSwap Nodes that have failed to load will show as red on the graph."?
> 
> How to do it?

This was the fix for this error:

>ImportError: cannot import name 'preserve_channel_dim' from 'albucore.utils'

Just update ReActor and run install.bat or delete ReActor and re-install it again

**Comment by DavidManas** [comment]
> > now i saw this: Cannot import C:\Users\ivayl\Desktop\ComfyUI\ComfyUI\custom_nodes\comfyui-reactor-node module for custom nodes: No module named 'insightface'
> 
> You have to go back to the main github page, there is a section in troubleshooting with the title
> 
> "I. (For Windows users) If you still cannot build Insightface for some reasons or just don't want to install Visual Studio or VS C++ Build Tools - do the following:"
> 
> You have to do what the author said, look at the version of python your reactor node local repo uses, I specify this because it may be different from the one you'd use, then download either insightface 3.10, 3.11, or 3.12 depending on the version that was printed in the console, then you will execute the wheel, make sure to restart comfyUI if it's running, it should solve the issue.

It doesn't work for me, I have installed the version that I was asked to install from the node manager, in this case insightface==0.7.3, and nothing, when I do: Building wheel for insightface, it gives the error which I have referenced in another post: https://github.com/Gourieff/comfyui-reactor-node/discussions/446
